### PR TITLE
[firebase_performance] Java side of FirebasePerformance rewrite

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

Possible reformat of FirebasePerformance plugin

## Related Issues

Dart: https://github.com/flutter/plugins/pull/1530
iOS: https://github.com/flutter/plugins/pull/1556

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_performance/android/src/main/java/io/flutter/plugins/firebaseperformance/FlutterFirebasePerformance.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// found in the LICENSE file.
+
+package io.flutter.plugins.firebaseperformance;
+
+import com.google.firebase.perf.FirebasePerformance;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+
+public class FlutterFirebasePerformance implements MethodChannel.MethodCallHandler {
```
[the remark this answers is not in the document]
Comment (bparrishMines) [reply packages/firebase_performance/android/src/main/java/io/flutter/plugins/firebaseperformance/FlutterFirebasePerformance.java]: No, I only separated this part as the FirebasePerformance plugin doesn't have any other classes that you can instantiate on there own. (e.g. to create an object you use `FirebasePeformance.newTrace()` or `FirebasePeformance.newHttpMetric()`. ML Kit works the same way, but Firebase Dynamic Links doesn't. 

I created this PR to get feedback on creating separate `MethodChannel`s for each class. Although its require more code, i think it's easier to read/maintain.

**Comment by bparrishMines** [comment]
Thats a good point. I'll add deregistration.

File: packages/firebase_performance/android/src/main/java/io/flutter/plugins/firebaseperformance/FlutterFirebasePerformance.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (collinjackson) [reply packages/firebase_performance/android/src/main/java/io/flutter/plugins/firebaseperformance/FlutterFirebasePerformance.java]: I'm on board with splitting things into more classes, generally. I'm ok with using it for this plugin as a one-off and we can discuss offline if we want to adopt the approach for plugins generally.

**Comment by bparrishMines** [comment]
Closing in favor of https://github.com/flutter/plugins/pull/1557

**Pull request closed without merging**